src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -2041,5 +2041,6 @@ xclCloseIPInterruptNotify(xclDeviceHandle handle, int fd)
 void
 xclGetDebugIpLayout(xclDeviceHandle hdl, char* buffer, size_t size, size_t* size_ret)
 {
+  *size_ret = 0;
```

Comment:
This code assumes that the `size_ret` pointer will always be valid.  To prevent future issues, could it first be checked for a null pointer.  For example:
```
if (size_ret != nullptr)
  *size_ret = 0;
```

Reply:
Added the change.